### PR TITLE
feat(minor-voting-verifier)!: verify msg ids

### PR DESCRIPTION
**Pull request by cjcobb23**
https://axelarnetwork.atlassian.net/browse/AXE-3836

deployment script change: https://github.com/axelarnetwork/axelar-contract-deployments/pull/219

### CODE REVIEW

File: contracts/voting-verifier/src/events.rs [hunk 1]
```diff
@@ -119,9 +120,33 @@ pub struct WorkerSetConfirmation {
     pub operators: Operators,
 }
 
+/// If parsing is successful, returns (tx_id, event_index). Otherwise returns ContractError::InvalidMessageID
+fn parse_message_id(
```
Comment (fish-sammy) [inline contracts/voting-verifier/src/events.rs:124]: Why not define such a method for `MessageIdFormat`?

File: contracts/voting-verifier/src/events.rs [hunk 2]
```diff
@@ -145,55 +170,20 @@ pub struct TxEventConfirmation {
...
-}
-
-pub fn construct_message_id(tx_hash: Hash, event_index: u32) -> String {
-    format!(
-        "{}{}{}",
-        HexBinary::from(tx_hash),
-        TX_HASH_EVENT_INDEX_SEPARATOR,
-        event_index
-    )
+pub fn make_tx_event_confirmation(
```
Comment (fish-sammy) [inline contracts/voting-verifier/src/events.rs:173]: Consider `impl TryFrom<(Message, &MessageIdFormat)> for TxEventConfirmation`

File: contracts/voting-verifier/src/events.rs (hunk 1 above)
Comment (cjcobb23) [reply contracts/voting-verifier/src/events.rs:124]: this method returns tx_id and event_index. Not every message id format has those two components. They all implement FromStr though.

File: contracts/voting-verifier/src/events.rs (hunk 2 above)
Comment (cjcobb23) [reply contracts/voting-verifier/src/events.rs:173]: I thought of that, it just felt a bit messy. But if you think that's cleaner, I can reconsider.

File: contracts/voting-verifier/src/events.rs (hunk 1 above)
Comment (fish-sammy) [reply contracts/voting-verifier/src/events.rs:124]: Sure, but it's not like you can ignore those that don't contain those two components in this function. Rust forces you to have all the options in the match block. You'd have to put `unimplemented!()` here anyway. I feel a method on the `MessageIdFormat` is a better indicator that it won't be implemented ever.

**Review by fish-sammy — APPROVED** [review]

File: contracts/voting-verifier/src/events.rs (hunk 1 above)
Comment (cjcobb23) [reply contracts/voting-verifier/src/events.rs:124]: I just don't like the idea of adding something to `MessageIdFormat` that is not generic. I think if we add formats that can't be parsed to this structure (string, uint), then we need to decide whether this voting verifier will support those formats. If it does, we change this function. If it doesn't, this function errors or panics. I'd rather address that if/when the time comes.

**Pull request merged**